src/main/scala/goatrodeo/util/FileWalker.scala

```diff
@@ -161,52 +266,57 @@ object FileWalker {
         }
       }
     }.toOption.flatten
+  }
+
+  /** Try a series of strategies (except for uncompressing a file) for creating
+    * an archive stream
+    *
+    * @param in
+    * @return
+    */
+  def tryToConstructArchiveStream(
+      in: ArtifactWrapper
+  ): OptionalArchiveStream = {
+    asZipContainer(in) orElse
```

Comment:
We might want to pull these out into implementations of an "archive stream wrapper" trait.  This would be useful if users want to bring their own implementation.

Reply:
Please see my above comment about the issues with the many formats.

We also need to think about how and where to create plugin points... and the plugins should be Java-friendly so folks can do plugins in Java... that may mean exposing Java not Scala collections and other fun and games.